includes/settings.php

```diff
@@ -454,3 +456,91 @@ function sanitize_settings( $settings ) {
 
 	return $new_settings;
 }
+
+/**
+ * Enable auto-update of plugin.
+ *
+ * @param stdClass $transient Transient cache object.
+ *
+ * @return stdClass
+ */
+function auto_update_plugins_setting( $transient ) {
+	if ( ! DT_IS_NETWORK ) {
+		$valid_license = \Distributor\Utils\get_settings()['valid_license'];
+	} else {
+		$valid_license = \Distributor\Utils\get_network_settings()['valid_license'];
+	}
+
+	if ( ! $valid_license ) {
+		return $transient;
+	}
+
+	$current_version = defined( 'DT_VERSION' ) ? DT_VERSION : null;
+	$update = check_for_updates( $current_version );
+	if ( $update ) {
+		$transient->response['distributor/distributor.php'] = $update;
+	} else {
+		$item = (object) array(
+			'id'            => 'distributor/distributor.php',
+			'slug'          => 'distributor',
+			'plugin'        => 'distributor/distributor.php',
+			'new_version'   => $current_version,
+			'url'           => 'https://github.com/10up/distributor',
+			'package'       => '',
+			'icons'         => array(),
+			'banners'       => array(),
+			'banners_rtl'   => array(),
+			'tested'        => '',
+			'requires_php'  => '',
+			'compatibility' => new \stdClass(),
+		);
+
+		$transient->no_update['distributor/distributor.php'] = $item;
+	}
+
+	return $transient;
+}
+
+/**
+ * Compare latest release version from plugin with the current installed version.
+ *
+ * @param string $current_version Current installed version.
+ *
+ * @return false|object
+ */
+function check_for_updates( $current_version ) {
```

Comment:
So I know we bundle the Plugin Update Checker and use that to check for normal updates. Curious to know if we could utilize that here instead of having to maintain our own function? I think it ends up doing mostly the same as this function does, as far as the remote request goes, so may not provide any value but would keep things inline to how we handle normal updates.